runtime/vm/JVMImage.cpp

```diff
@@ -342,7 +395,7 @@ JVMImage::readImageFromFile(void)
 	_jvmImageHeader = (JVMImageHeader *)mmap(
 		(void *)imageHeaderBuffer.imageAlignedAddress,
 		imageHeaderBuffer.imageSize,
-		PROT_READ, MAP_PRIVATE | MAP_FIXED, fileDescriptor, 0);
+		PROT_READ | PROT_WRITE, MAP_SHARED | MAP_FIXED, fileDescriptor, 0);
```

Comment:
Not sure if we want to use MAP_SHARED. Maybe we want to keep the writing separate.

Because classes loaded during warm runs might not be completely deterministic (due to anonymouse classes and other in-determinism). In a warm run it is quite likely that you will need to load a small # of classes. These classes may not be the same from run to run so we shouldn't pollute the image.

Reply:
Yeah I need to take it out